In [3]:
import pandas as pd
import numpy as np
import matplotlib as plt


# Now we load the csv

In [16]:
dataframe = pd.read_csv("loan.csv",low_memory=False)
dataframe.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501    1296599       5000         5000           4975.0   36 months   
1  1077430    1314167       2500         2500           2500.0   60 months   
2  1077175    1313524       2400         2400           2400.0   36 months   
3  1076863    1277178      10000        10000          10000.0   36 months   
4  1075358    1311748       3000         3000           3000.0   60 months   

  int_rate  installment grade sub_grade  ... num_tl_90g_dpd_24m  \
0   10.65%       162.87     B        B2  ...                NaN   
1   15.27%        59.83     C        C4  ...                NaN   
2   15.96%        84.33     C        C5  ...                NaN   
3   13.49%       339.31     C        C1  ...                NaN   
4   12.69%        67.79     B        B5  ...                NaN   

  num_tl_op_past_12m pct_tl_nvr_dlq  percent_bc_gt_75 pub_rec_bankruptcies  \
0                NaN            NaN               NaN                  0.0   
1                NaN            NaN               NaN                  0.0   
2                NaN            NaN               NaN                  0.0   
3                NaN            NaN               NaN                  0.0   
4                NaN            NaN               NaN                  0.0   

  tax_liens tot_hi_cred_lim total_bal_ex_mort total_bc_limit  \
0       0.0             NaN               NaN            NaN   
1       0.0             NaN               NaN            NaN   
2       0.0             NaN               NaN            NaN   
3       0.0             NaN               NaN            NaN   
4       0.0             NaN               NaN            NaN   

  total_il_high_credit_limit  
0                        NaN  
1                        NaN  
2                        NaN  
3                        NaN  
4                        NaN  

[5 rows x 111 columns]

In [17]:
dataframe.describe()


id     member_id     loan_amnt   funded_amnt  \
count  3.971700e+04  3.971700e+04  39717.000000  39717.000000   
mean   6.831319e+05  8.504636e+05  11219.443815  10947.713196   
std    2.106941e+05  2.656783e+05   7456.670694   7187.238670   
min    5.473400e+04  7.069900e+04    500.000000    500.000000   
25%    5.162210e+05  6.667800e+05   5500.000000   5400.000000   
50%    6.656650e+05  8.508120e+05  10000.000000   9600.000000   
75%    8.377550e+05  1.047339e+06  15000.000000  15000.000000   
max    1.077501e+06  1.314167e+06  35000.000000  35000.000000   

       funded_amnt_inv   installment    annual_inc           dti  \
count     39717.000000  39717.000000  3.971700e+04  39717.000000   
mean      10397.448868    324.561922  6.896893e+04     13.315130   
std        7128.450439    208.874874  6.379377e+04      6.678594   
min           0.000000     15.690000  4.000000e+03      0.000000   
25%        5000.000000    167.020000  4.040400e+04      8.170000   
50%        8975.000000    280.220000  5.900000e+04     13.400000   
75%       14400.000000    430.780000  8.230000e+04     18.600000   
max       35000.000000   1305.190000  6.000000e+06     29.990000   

        delinq_2yrs  inq_last_6mths  ...  num_tl_90g_dpd_24m  \
count  39717.000000    39717.000000  ...                 0.0   
mean       0.146512        0.869200  ...                 NaN   
std        0.491812        1.070219  ...                 NaN   
min        0.000000        0.000000  ...                 NaN   
25%        0.000000        0.000000  ...                 NaN   
50%        0.000000        1.000000  ...                 NaN   
75%        0.000000        1.000000  ...                 NaN   
max       11.000000        8.000000  ...                 NaN   

       num_tl_op_past_12m  pct_tl_nvr_dlq  percent_bc_gt_75  \
count                 0.0             0.0               0.0   
mean                  NaN             NaN               NaN   
std                   NaN             NaN               NaN   
min                   NaN             NaN               NaN   
25%                   NaN             NaN               NaN   
50%                   NaN             NaN               NaN   
75%                   NaN             NaN               NaN   
max                   NaN             NaN               NaN   

       pub_rec_bankruptcies  tax_liens  tot_hi_cred_lim  total_bal_ex_mort  \
count          39020.000000    39678.0              0.0                0.0   
mean               0.043260        0.0              NaN                NaN   
std                0.204324        0.0              NaN                NaN   
min                0.000000        0.0              NaN                NaN   
25%                0.000000        0.0              NaN                NaN   
50%                0.000000        0.0              NaN                NaN   
75%                0.000000        0.0              NaN                NaN   
max                2.000000        0.0              NaN                NaN   

       total_bc_limit  total_il_high_credit_limit  
count             0.0                         0.0  
mean              NaN                         NaN  
std               NaN                         NaN  
min               NaN                         NaN  
25%               NaN                         NaN  
50%               NaN                         NaN  
75%               NaN                         NaN  
max               NaN                         NaN  

[8 rows x 87 columns]

In [18]:
print(len(dataframe))


39717


In [19]:
# check null values for all columns 
dataframe.isna().sum()

id                                0
member_id                         0
loan_amnt                         0
funded_amnt                       0
funded_amnt_inv                   0
                              ...  
tax_liens                        39
tot_hi_cred_lim               39717
total_bal_ex_mort             39717
total_bc_limit                39717
total_il_high_credit_limit    39717
Length: 111, dtype: int64

We observe that 39717 is total count we have null values for many columns having all null values
No use using them , hence we will remove them 


In [96]:
dataframe_colremoved = dataframe.drop(columns=['tot_hi_cred_lim','total_bal_ex_mort','total_bc_limit',
'total_il_high_credit_limit','num_tl_op_past_12m','pct_tl_nvr_dlq','percent_bc_gt_75','num_tl_120dpd_2m',
'num_tl_30dpd','num_tl_90g_dpd_24m','num_rev_accts','num_rev_tl_bal_gt_0','num_sats','num_bc_tl','num_il_tl',
'num_op_rev_tl','num_actv_bc_tl','num_actv_rev_tl','num_bc_sats','mths_since_recent_inq',
'mths_since_recent_revol_delinq','num_accts_ever_120_pd','mort_acc','mths_since_recent_bc','mths_since_recent_bc_dlq',
'mo_sin_old_rev_tl_op','mo_sin_rcnt_rev_tl_op','mo_sin_rcnt_tl','mo_sin_old_il_acct','bc_util','bc_open_to_buy',
'avg_cur_bal','acc_open_past_24mths','inq_last_12m','total_bal_il','open_rv_12m','annual_inc_joint',
'dti_joint','verification_status_joint','tot_coll_amt','tot_cur_bal','open_acc_6m','open_il_6m',
'open_il_12m','open_il_24m','mths_since_rcnt_il','il_util','open_rv_24m','mths_since_last_major_derog',
'all_util','inq_last_12m','total_cu_tl','max_bal_bc','total_rev_hi_lim','inq_fi'])

# These 2 columns have a little less record than all hence we will retain them 
#,'next_pymnt_d','mths_since_last_record'

#'id', 'member_id', 'loan_amnt', 'funded_amnt', 'funded_amnt_inv', 'term', 'int_rate', 'installment', 'grade', 'sub_grade', 
# 'emp_title', 'emp_length', 'home_ownership', 'annual_inc', 'verification_status', 'issue_d', 'loan_status', 'pymnt_plan', 
# 'url', 'desc', 'purpose', 'title', 'zip_code', 'addr_state', 'dti', 'delinq_2yrs', 'earliest_cr_line', 'inq_last_6mths', 
# 'mths_since_last_delinq', 'mths_since_last_record', 'open_acc', 'pub_rec', 'revol_bal', 'revol_util', 'total_acc', 
# 'initial_list_status', 'out_prncp', 'out_prncp_inv', 'total_pymnt', 'total_pymnt_inv', 'total_rec_prncp', 'total_rec_int', 
# 'total_rec_late_fee', 'recoveries', 'collection_recovery_fee', 'last_pymnt_d', 'last_pymnt_amnt',
#  'next_pymnt_d', 'last_credit_pull_d', 'collections_12_mths_ex_med', 'policy_code', 'application_type', 
# 'acc_now_delinq', 'chargeoff_within_12_mths', 'delinq_amnt', 'pub_rec_bankruptcies', 'tax_liens'


In [97]:

dataframe_colremoved.head()


id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501    1296599       5000         5000           4975.0   36 months   
1  1077430    1314167       2500         2500           2500.0   60 months   
2  1077175    1313524       2400         2400           2400.0   36 months   
3  1076863    1277178      10000        10000          10000.0   36 months   
4  1075358    1311748       3000         3000           3000.0   60 months   

  int_rate  installment grade sub_grade  ... next_pymnt_d last_credit_pull_d  \
0   10.65%       162.87     B        B2  ...          NaN             May-16   
1   15.27%        59.83     C        C4  ...          NaN             Sep-13   
2   15.96%        84.33     C        C5  ...          NaN             May-16   
3   13.49%       339.31     C        C1  ...          NaN             Apr-16   
4   12.69%        67.79     B        B5  ...       Jun-16             May-16   

  collections_12_mths_ex_med  policy_code application_type acc_now_delinq  \
0                        0.0            1       INDIVIDUAL              0   
1                        0.0            1       INDIVIDUAL              0   
2                        0.0            1       INDIVIDUAL              0   
3                        0.0            1       INDIVIDUAL              0   
4                        0.0            1       INDIVIDUAL              0   

  chargeoff_within_12_mths delinq_amnt pub_rec_bankruptcies tax_liens  
0                      0.0           0                  0.0       0.0  
1                      0.0           0                  0.0       0.0  
2                      0.0           0                  0.0       0.0  
3                      0.0           0                  0.0       0.0  
4                      0.0           0                  0.0       0.0  

[5 rows x 57 columns]

In [73]:
dataframe_colremoved2 = dataframe.dropna(axis=1,how='all')


In [98]:
dataframe_colremoved2.shape


(39717, 57)

In [76]:
print(list(dataframe_colremoved2))

['id', 'member_id', 'loan_amnt', 'funded_amnt', 'funded_amnt_inv', 'term', 'int_rate', 'installment', 'grade', 'sub_grade', 'emp_title', 'emp_length', 'home_ownership', 'annual_inc', 'verification_status', 'issue_d', 'loan_status', 'pymnt_plan', 'url', 'desc', 'purpose', 'title', 'zip_code', 'addr_state', 'dti', 'delinq_2yrs', 'earliest_cr_line', 'inq_last_6mths', 'mths_since_last_delinq', 'mths_since_last_record', 'open_acc', 'pub_rec', 'revol_bal', 'revol_util', 'total_acc', 'initial_list_status', 'out_prncp', 'out_prncp_inv', 'total_pymnt', 'total_pymnt_inv', 'total_rec_prncp', 'total_rec_int', 'total_rec_late_fee', 'recoveries', 'collection_recovery_fee', 'last_pymnt_d', 'last_pymnt_amnt', 'next_pymnt_d', 'last_credit_pull_d', 'collections_12_mths_ex_med', 'policy_code', 'application_type', 'acc_now_delinq', 'chargeoff_within_12_mths', 'delinq_amnt', 'pub_rec_bankruptcies', 'tax_liens']


In [99]:
#print(dataframe_colremoved.isna().sum())
#check the dataframe now 
dataframe_colremoved.isnull().sum().sort_values(ascending = False).head(50)


next_pymnt_d                  38577
mths_since_last_record        36931
mths_since_last_delinq        25682
desc                          12940
emp_title                      2459
emp_length                     1075
pub_rec_bankruptcies            697
last_pymnt_d                     71
chargeoff_within_12_mths         56
collections_12_mths_ex_med       56
revol_util                       50
tax_liens                        39
title                            11
last_credit_pull_d                2
home_ownership                    0
int_rate                          0
out_prncp_inv                     0
total_pymnt                       0
total_pymnt_inv                   0
total_rec_prncp                   0
total_rec_int                     0
total_rec_late_fee                0
recoveries                        0
collection_recovery_fee           0
term                              0
last_pymnt_amnt                   0
initial_list_status               0
funded_amnt_inv             

In [115]:
# 11 values have null for Title , we will replace them with NA
dataframe_colremoved.title.unique()

dataframe_colremoved["title"].fillna("NA", inplace=True)

dataframe_colremoved["emp_title"].fillna("NA", inplace=True)


In [116]:
dataframe_colremoved.isna().sum().sort_values(ascending = False).head(50)

next_pymnt_d                  38577
mths_since_last_record        36931
mths_since_last_delinq        25682
desc                          12940
emp_length                     1075
pub_rec_bankruptcies            697
last_pymnt_d                     71
chargeoff_within_12_mths         56
collections_12_mths_ex_med       56
revol_util                       50
tax_liens                        39
last_credit_pull_d                2
emp_title                         0
int_rate                          0
out_prncp                         0
out_prncp_inv                     0
total_pymnt                       0
total_pymnt_inv                   0
total_rec_prncp                   0
total_rec_int                     0
total_rec_late_fee                0
recoveries                        0
collection_recovery_fee           0
last_pymnt_amnt                   0
home_ownership                    0
term                              0
funded_amnt_inv                   0
policy_code                 

In [109]:
pd.set_option("display.max_columns", 200)


In [110]:
dataframe_colremoved.head()


id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501    1296599       5000         5000           4975.0   36 months   
1  1077430    1314167       2500         2500           2500.0   60 months   
2  1077175    1313524       2400         2400           2400.0   36 months   
3  1076863    1277178      10000        10000          10000.0   36 months   
4  1075358    1311748       3000         3000           3000.0   60 months   

  int_rate  installment grade sub_grade                 emp_title emp_length  \
0   10.65%       162.87     B        B2                        NA  10+ years   
1   15.27%        59.83     C        C4                     Ryder   < 1 year   
2   15.96%        84.33     C        C5                        NA  10+ years   
3   13.49%       339.31     C        C1       AIR RESOURCES BOARD  10+ years   
4   12.69%        67.79     B        B5  University Medical Group     1 year   

  home_ownership  annual_inc verification_status issue_d  loan_status  \
0           RENT     24000.0            Verified  Dec-11   Fully Paid   
1           RENT     30000.0     Source Verified  Dec-11  Charged Off   
2           RENT     12252.0        Not Verified  Dec-11   Fully Paid   
3           RENT     49200.0     Source Verified  Dec-11   Fully Paid   
4           RENT     80000.0     Source Verified  Dec-11      Current   

  pymnt_plan                                                url  \
0          n  https://lendingclub.com/browse/loanDetail.acti...   
1          n  https://lendingclub.com/browse/loanDetail.acti...   
2          n  https://lendingclub.com/browse/loanDetail.acti...   
3          n  https://lendingclub.com/browse/loanDetail.acti...   
4          n  https://lendingclub.com/browse/loanDetail.acti...   

                                                desc         purpose  \
0    Borrower added on 12/22/11 > I need to upgra...     credit_card   
1    Borrower added on 12/22/11 > I plan to use t...             car   
2                                                NaN  small_business   
3    Borrower added on 12/21/11 > to pay for prop...           other   
4    Borrower added on 12/21/11 > I plan on combi...           other   

                  title zip_code addr_state    dti  delinq_2yrs  \
0              Computer    860xx         AZ  27.65            0   
1                  bike    309xx         GA   1.00            0   
2  real estate business    606xx         IL   8.72            0   
3              personel    917xx         CA  20.00            0   
4              Personal    972xx         OR  17.94            0   

  earliest_cr_line  inq_last_6mths  mths_since_last_delinq  \
0           Jan-85               1                     NaN   
1           Apr-99               5                     NaN   
2           Nov-01               2                     NaN   
3           Feb-96               1                    35.0   
4           Jan-96               0                    38.0   

   mths_since_last_record  open_acc  pub_rec  revol_bal revol_util  total_acc  \
0                     NaN         3        0      13648     83.70%          9   
1                     NaN         3        0       1687      9.40%          4   
2                     NaN         2        0       2956     98.50%         10   
3                     NaN        10        0       5598        21%         37   
4                     NaN        15        0      27783     53.90%         38   

  initial_list_status  out_prncp  out_prncp_inv   total_pymnt  \
0                   f       0.00           0.00   5863.155187   
1                   f       0.00           0.00   1008.710000   
2                   f       0.00           0.00   3005.666844   
3                   f       0.00           0.00  12231.890000   
4                   f     524.06         524.06   3513.330000   

   total_pymnt_inv  total_rec_prncp  total_rec_int  total_rec_late_fee  \
0          5833.84          5000.00         863.16     

In [113]:
dataframe_colremoved.shape


(39717, 57)

In [122]:
# getting count of unique values 
dataframe_colremoved.nunique(axis=0).sort_values().head(20).index


Index(['tax_liens', 'delinq_amnt', 'chargeoff_within_12_mths',
       'acc_now_delinq', 'application_type', 'policy_code',
       'collections_12_mths_ex_med', 'pymnt_plan', 'initial_list_status',
       'term', 'next_pymnt_d', 'pub_rec_bankruptcies', 'loan_status',
       'verification_status', 'pub_rec', 'home_ownership', 'grade',
       'inq_last_6mths', 'emp_length', 'delinq_2yrs'],
      dtype='object')

In [151]:
#dataframe_colremoved.drop([dataframe_colremoved.nunique(axis=0).sort_values().head(9).index],axis=1)
#dataframe_colremoved.columns.to_list

dataframe_coluniq = dataframe_colremoved.drop(dataframe_colremoved.nunique(axis=0).sort_values().head(9).index,axis=1)

#confirm columns with Single value removed
dataframe_coluniq.nunique(axis=0).sort_values().head(57)


next_pymnt_d                   2
term                           2
pub_rec_bankruptcies           3
loan_status                    3
verification_status            3
home_ownership                 5
pub_rec                        5
grade                          7
inq_last_6mths                 9
emp_length                    11
delinq_2yrs                   11
purpose                       14
sub_grade                     35
open_acc                      40
addr_state                    50
issue_d                       55
total_acc                     82
mths_since_last_delinq        95
last_pymnt_d                 101
last_credit_pull_d           106
mths_since_last_record       111
int_rate                     371
earliest_cr_line             526
zip_code                     823
loan_amnt                    885
funded_amnt                 1041
revol_util                  1089
out_prncp                   1137
out_prncp_inv               1138
total_rec_late_fee          1356
collection

In [150]:
dataframe_coluniq.shape






(39717, 48)

In [160]:
print(dataframe_coluniq[['id','url']])

            id                                                url
0      1077501  https://lendingclub.com/browse/loanDetail.acti...
1      1077430  https://lendingclub.com/browse/loanDetail.acti...
2      1077175  https://lendingclub.com/browse/loanDetail.acti...
3      1076863  https://lendingclub.com/browse/loanDetail.acti...
4      1075358  https://lendingclub.com/browse/loanDetail.acti...
...        ...                                                ...
39712    92187  https://lendingclub.com/browse/loanDetail.acti...
39713    90665  https://lendingclub.com/browse/loanDetail.acti...
39714    90395  https://lendingclub.com/browse/loanDetail.acti...
39715    90376  https://lendingclub.com/browse/loanDetail.acti...
39716    87023  https://lendingclub.com/browse/loanDetail.acti...

[39717 rows x 2 columns]


Makes no sense to keep the columns id and url , so remove them 

In [161]:
dataframe_coluniq.drop(['id','url'],axis=1,inplace=True)
dataframe_coluniq


member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0        1296599       5000         5000           4975.0   36 months   
1        1314167       2500         2500           2500.0   60 months   
2        1313524       2400         2400           2400.0   36 months   
3        1277178      10000        10000          10000.0   36 months   
4        1311748       3000         3000           3000.0   60 months   
...          ...        ...          ...              ...         ...   
39712      92174       2500         2500           1075.0   36 months   
39713      90607       8500         8500            875.0   36 months   
39714      90390       5000         5000           1325.0   36 months   
39715      89243       5000         5000            650.0   36 months   
39716      86999       7500         7500            800.0   36 months   

      int_rate  installment grade sub_grade                   emp_title  \
0       10.65%       162.87     B        B2                          NA   
1       15.27%        59.83     C        C4                       Ryder   
2       15.96%        84.33     C        C5                          NA   
3       13.49%       339.31     C        C1         AIR RESOURCES BOARD   
4       12.69%        67.79     B        B5    University Medical Group   
...        ...          ...   ...       ...                         ...   
39712    8.07%        78.42     A        A4             FiSite Research   
39713   10.28%       275.38     C        C1  Squarewave Solutions, Ltd.   
39714    8.07%       156.84     A        A4                          NA   
39715    7.43%       155.38     A        A2                          NA   
39716   13.75%       255.43     E        E2            Evergreen Center   

      emp_length home_ownership  annual_inc verification_status issue_d  \
0      10+ years           RENT     24000.0            Verified  Dec-11   
1       < 1 year           RENT     30000.0     Source Verified  Dec-11   
2      10+ years           RENT     12252.0        Not Verified  Dec-11   
3      10+ years           RENT     49200.0     Source Verified  Dec-11   
4         1 year           RENT     80000.0     Source Verified  Dec-11   
...          ...            ...         ...                 ...     ...   
39712    4 years       MORTGAGE    110000.0        Not Verified  Jul-07   
39713    3 years           RENT     18000.0        Not Verified  Jul-07   
39714   < 1 year       MORTGAGE    100000.0        Not Verified  Jul-07   
39715   < 1 year       MORTGAGE    200000.0        Not Verified  Jul-07   
39716   < 1 year            OWN     22000.0        Not Verified  Jun-07   

       loan_status                                               desc  \
0       Fully Paid    Borrower added on 12/22/11 > I need to upgra...   
1      Charged Off    Borrower added on 12/22/11 > I plan to use t...   
2       Fully Paid                                                NaN   
3       Fully Paid    Borrower added on 12/21/11 > to pay for prop...   
4          Current    Borrower added on 12/21/11 > I plan on combi...   
...            ...                                                ...   
39712   Fully Paid  Our current gutter system on our home is old a...   
39713   Fully Paid  The rate of interest and fees incurred by carr...   
39714   Fully Paid                                                NaN   
39715   Fully Paid                                                NaN   
39716   Fully Paid  I plan to consolidate over $7,000 of debt: a c...   

                  purpose                      title zip_code addr_state  \
0             credit_card                   Computer    860xx         AZ   
1                     car                       bike    309xx         GA   
2          small_business       real estate business    606xx         IL   
3                   other                   personel    917xx         CA   
4                   other                   Personal    972xx         O

In [162]:
dataframe_coluniq.grade.unique()


array(['B', 'C', 'A', 'E', 'F', 'D', 'G'], dtype=object)

In [163]:
#No use keeping employee title 
dataframe_coluniq.drop(['emp_title'],axis=1,inplace=True)
dataframe_coluniq

member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0        1296599       5000         5000           4975.0   36 months   
1        1314167       2500         2500           2500.0   60 months   
2        1313524       2400         2400           2400.0   36 months   
3        1277178      10000        10000          10000.0   36 months   
4        1311748       3000         3000           3000.0   60 months   
...          ...        ...          ...              ...         ...   
39712      92174       2500         2500           1075.0   36 months   
39713      90607       8500         8500            875.0   36 months   
39714      90390       5000         5000           1325.0   36 months   
39715      89243       5000         5000            650.0   36 months   
39716      86999       7500         7500            800.0   36 months   

      int_rate  installment grade sub_grade emp_length home_ownership  \
0       10.65%       162.87     B        B2  10+ years           RENT   
1       15.27%        59.83     C        C4   < 1 year           RENT   
2       15.96%        84.33     C        C5  10+ years           RENT   
3       13.49%       339.31     C        C1  10+ years           RENT   
4       12.69%        67.79     B        B5     1 year           RENT   
...        ...          ...   ...       ...        ...            ...   
39712    8.07%        78.42     A        A4    4 years       MORTGAGE   
39713   10.28%       275.38     C        C1    3 years           RENT   
39714    8.07%       156.84     A        A4   < 1 year       MORTGAGE   
39715    7.43%       155.38     A        A2   < 1 year       MORTGAGE   
39716   13.75%       255.43     E        E2   < 1 year            OWN   

       annual_inc verification_status issue_d  loan_status  \
0         24000.0            Verified  Dec-11   Fully Paid   
1         30000.0     Source Verified  Dec-11  Charged Off   
2         12252.0        Not Verified  Dec-11   Fully Paid   
3         49200.0     Source Verified  Dec-11   Fully Paid   
4         80000.0     Source Verified  Dec-11      Current   
...           ...                 ...     ...          ...   
39712    110000.0        Not Verified  Jul-07   Fully Paid   
39713     18000.0        Not Verified  Jul-07   Fully Paid   
39714    100000.0        Not Verified  Jul-07   Fully Paid   
39715    200000.0        Not Verified  Jul-07   Fully Paid   
39716     22000.0        Not Verified  Jun-07   Fully Paid   

                                                    desc             purpose  \
0        Borrower added on 12/22/11 > I need to upgra...         credit_card   
1        Borrower added on 12/22/11 > I plan to use t...                 car   
2                                                    NaN      small_business   
3        Borrower added on 12/21/11 > to pay for prop...               other   
4        Borrower added on 12/21/11 > I plan on combi...               other   
...                                                  ...                 ...   
39712  Our current gutter system on our home is old a...    home_improvement   
39713  The rate of interest and fees incurred by carr...         credit_card   
39714                                                NaN  debt_consolidation   
39715                                                NaN               other   
39716  I plan to consolidate over $7,000 of debt: a c...  debt_consolidation   

                           title zip_code addr_state    dti  delinq_2yrs  \
0                       Computer    860xx         AZ  27.65            0   
1                           bike    309xx         GA   1.00            0   
2           real estate business    606xx         IL   8.72            0   
3                       personel    917xx         CA  20.00            0   
4                       Personal    972xx         OR  17.94            0   
...                          ...      ...        ...    ...          ...   
39712          

In [167]:
dataframe_coluniq.home_ownership.unique()
#dataframe_coluniq.home_ownership.value_counts.sum()
dataframe_coluniq.groupby('home_ownership').count()


member_id  loan_amnt  funded_amnt  funded_amnt_inv   term  \
home_ownership                                                              
MORTGAGE            17659      17659        17659            17659  17659   
NONE                    3          3            3                3      3   
OTHER                  98         98           98               98     98   
OWN                  3058       3058         3058             3058   3058   
RENT                18899      18899        18899            18899  18899   

                int_rate  installment  grade  sub_grade  emp_length  \
home_ownership                                                        
MORTGAGE           17659        17659  17659      17659       17212   
NONE                   3            3      3          3           3   
OTHER                 98           98     98         98          98   
OWN                 3058         3058   3058       3058        2841   
RENT               18899        18899  18899      18899       18488   

                annual_inc  verification_status  issue_d  loan_status   desc  \
home_ownership                                                                 
MORTGAGE             17659                17659    17659        17659  11822   
NONE                     3                    3        3            3      3   
OTHER                   98                   98       98           98     98   
OWN                   3058                 3058     3058         3058   2021   
RENT                 18899                18899    18899        18899  12833   

                purpose  title  zip_code  addr_state    dti  delinq_2yrs  \
home_ownership                                                             
MORTGAGE          17659  17659     17659       17659  17659        17659   
NONE                  3      3         3           3      3            3   
OTHER                98     98        98          98     98           98   
OWN                3058   3058      3058        3058   3058         3058   
RENT              18899  18899     18899       18899  18899        18899   

                earliest_cr_line  inq_last_6mths  mths_since_last_delinq  \
home_ownership                                                             
MORTGAGE                   17659           17659                    6360   
NONE                           3               3                       1   
OTHER                         98              98                      37   
OWN                         3058            3058                    1067   
RENT                       18899           18899                    6570   

                mths_since_last_record  open_acc  pub_rec  revol_bal  \
home_ownership                                                         
MORTGAGE                          1309     17659    17659      17659   
NONE                                 1         3        3          3   
OTHER                                4        98       98         98   
OWN                                210      3058     3058       3058   
RENT                              1262     18899    18899      18899   

                revol_util  total_acc  out_prncp  out_prncp_inv  total_pymnt  \
home_ownership                                                                 
MORTGAGE             17648      17659      17659          17659        17659   
NONE                     3          3          3              3            3   
OTHER                   96         98         98             98           98   
OWN                   3053       3058       3058           3058         3058   
RENT                 18867      18899      18899          18899        18899   

                total_pymnt_inv  total_rec_prncp  total_rec_int  \
home_ownership                                                    
MORTGAGE                  17659            17659          17659   
NONE                          3                3              3   
OTHER            

In [173]:
#remove rows with None 
dataframe_coluniq = dataframe_coluniq[dataframe_coluniq.home_ownership != 'NONE']

dataframe_coluniq.shape


(39714, 45)